lib/RWMutex.ts

```
@@ -65,10 +76,17 @@ export default class RWMutex {
           },
         });
         if (result.matchedCount > 0) {
+          if (typeof cb === "function") {
+            cb();
+          }
           return;
         }
       } catch (err) {
-        throw new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        const formattedError = new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        if (typeof cb === "function") {
+          cb(formattedError);
```

you probably want to `return cb ...` here and in a couple other places

---

I actually don't think it's necessary. My understanding is that the `throw` will be captured as a rejected promise because these functions are all tagged `async`. So if you call `lock.lock((err) => {...})`, you'll just be ignoring the Promise returned by the function and the `throw` is essentially a noop in that case.

I can add it if we think it's clearer though.

---

I don't think we want to throw the error though? If you call the function with a callback you get the error through the callback you don't then want an error to be thrown also?